native/e2e/recorders/android-recorder.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const { rawCommand, command, sleep } = require('./../utils');
+
+function video(filename) {
+  let handler;
+  function start() {
+    handler = rawCommand(
+      'adb',
+      'shell',
+      'screenrecord',
+      ' --bit-rate',
+      '6000000',
+      `/sdcard/${filename}.mp4`,
+    );
+  }
+  function stop() {
+    if (!handler) {
+      throw new Error(
+        'Cannot call stop video if the process has not being started',
+      );
+    }
+    handler.kill('SIGINT');
+    handler.stdin.write('\n');
+  }
+  async function get(path) {
+    return sleep(1000)
+      .then(command('adb', 'pull', `/sdcard/${filename}.mp4`, path))
+      .then(sleep(1000));
+  }
+  async function remove() {
+    return command('adb', 'shell', 'rm', '-f', `/sdcard/${filename}.mp4`);
+  }
+  return {
+    stop,
+    start,
+    get,
+    remove,
+  };
+}
+
+class Recorder {
+  constructor(filePath, fileName) {
+    this.filePath = filePath;
+    this.fileName = fileName;
+    this.handle = video(fileName);
+  }
+
+  async startRecording() {
+    this.handle.start();
+  }
+
+  async finishRecording() {
+    this.handle.stop();
+    await sleep(1000);
+    await this.handle
+      .get(this.filePath.replace(this.fileName, ''))
+      .then(() => this.handle.remove());
```

Comment:
As you're calling this with no arguments, I think you can use this instead:

`.then(this.handle.remove);`

Reply:
you can use point-free style but technically the return value of the previous end block will be passed as an argument to the remove function that does not accepts any, can lead to confusion when reading the code in the future

Reply:
Ah fair enough!